### PR TITLE
feat(cli): add --json output for status and data validate

### PR DESCRIPTION
**Pull request by yashksaini-coder**
Addresses #69. Deliberately not using a closing keyword — `benchmark --json` is still outstanding (see *Scope* below), so whether #69 closes on merge or stays open for that is your call.

## What

`--json` on `nanotune status` and `nanotune data validate`. Prints a single JSON document to stdout instead of the Ink report, so the read-only commands can be consumed by scripts and CI.

```bash
$ nanotune status --json | jq '.benchmarks.latest.passRate'
0.9

$ nanotune data validate --json | jq '.checks'
{
  "dataFileExists": true,
  "validJsonStructure": true,
  "contextMessageConsistency": true,
  "noDuplicateInputs": false,
  "minimumExampleCount": true
}
```

Ink never mounts in this mode. `render()` writes to stdout the moment it does, and a box-drawn frame in the middle of a JSON document is not something a consumer can recover from — so `cli.tsx` branches *before* rendering and calls a plain collector instead.

## The contract

Documented in [`docs/guides/json-output.md`](docs/guides/json-output.md), and what the schemas can be relied on against:

- **stdout carries the payload or nothing.** One JSON document plus a newline, or nothing at all. Never a diagnostic, never a partial document.
- **Diagnostics go to stderr** — including the unknown-config-key warnings from `loadConfig`, which already used `console.warn` and so were already correct.
- **The exit code carries the status**, unchanged from the Ink path. `data validate --json` still exits `1` on invalid data — and still prints its report, since the report is the useful part.
- **Absent values are `null`, never omitted**, so `jq` never has to tell a missing key from an empty one.
- **Timestamps are ISO 8601, sizes are raw bytes.** `2 hours ago` and `1.2 GB` are display concerns and stay in the view layer.

No `schemaVersion` field, deliberately: fields may be added in a minor release, and if a break ever becomes necessary a version field can be introduced then, with its absence read as v1. **Happy to add one up front if you'd rather** — it's the one call here I'd expect an opinion on.

## Shared model, not a second code path

`collectStatus()` and `collectValidation()` move the data gathering into `src/lib/`, and the Ink views consume the *same* functions. A `--json`-only code path would have let the two drift; this way they cannot. `status.tsx` and `validate.tsx` come out ~100 lines lighter and are now pure presentation.

## Two fixes that fall out of it

Both were required to ship a schema that can honestly be called stable:

1. **`data validate` was deriving its check verdicts by pattern-matching its own warning prose.**

   ```tsx
   status={!result.warnings.some(w => w.includes('duplicate')) ? 'success' : 'warning'}
   ```

   `validateTrainingData` already computed `duplicateCount` and `inconsistentPromptCount`, then discarded the numbers, stringified them into English, and had the view parse the English back out. Rewording a warning silently flipped a check. `ValidationResult` now carries the counts, and both views read those.

2. **The "minimum example count" check no longer applies to a validation set.** `validateTrainingData` already skips the 50-example warning for `--eval` ("a validation set is a slice of the training data"), and `docs/commands/data.md` already documents that — but the badge showed a warning anyway. Now consistent across the validator, the TUI and the JSON.

Both are behaviour changes, small and in service of the schema. Easy to drop if you'd rather they were separate.

## Scope

`benchmark --json` is **not** in this PR. Its run lives inside a ~550-line `useCallback` in the command component and wants extracting into a `lib/` async generator first — a refactor big enough to deserve its own review. Happy to pick it up as a follow-up. The four acceptance-criteria boxes on #69 are met by what's here.

`doctor --json` is out too, since #65 hasn't landed.

## Testing

21 new tests, 487 passing. `pnpm test:lint`, `test:format`, `test:types`, `test:knip`, `test:audit` and `test:ava` all green locally (semgrep not installed here — leaving that to CI).

`buildJsonOutcome` is split out from `emitJson` so the contract is testable without capturing real stdout, following the `buildTrainingArgs`/`runTraining` split.

Verified against the built binary in a scratch project:

| Check | Result |
|---|---|
| `nanotune status --json \| jq .` | ✅ |
| Unknown config key present → stdout still parses clean, warning on stderr | ✅ |
| Outside a project → 0 bytes on stdout, message on stderr, exit 1 | ✅ |
| `data validate --json` on invalid data → report printed, exit 1 | ✅ |
| `--fix` reports repairs and re-validates against what's left on disk | ✅ |
| `--eval` exempt from the 50-example floor | ✅ |
| Zero ANSI escape sequences in stdout (Ink genuinely never rendered) | ✅ |
| Interactive `nanotune status` output unchanged | ✅ |

### CODE REVIEW

**Comment by yashksaini-coder** [comment]
Follow-up is up: #158 adds `--json` to `nanotune benchmark`, which completes #69.

It's stacked on this branch, so its diff against `main` currently includes this PR's commit too — merging this one first reduces it to the benchmark commit alone.

Worth knowing before you review this one: extracting the benchmark run turned up the same class of issue the `ValidationResult` fix here addresses. A run whose llama-server dies mid-suite saves partial results and scores against the tests that actually ran — which made a saved partial run indistinguishable from a complete one with a worse pass rate. #158 adds an optional `warning` field to `BenchmarkResult` to close that. Same pattern both times: publishing a machine-readable schema is what forces the information loss into the open.

**Comment by yashksaini-coder** [comment]
@will-lamerton @akramcodez — ready for review whenever you have a moment.

Adds `--json` to `nanotune status` and `nanotune data validate` for #69. All checks green.

#158 is stacked on this one and covers `benchmark --json`; it shrinks to a single commit once this merges.
